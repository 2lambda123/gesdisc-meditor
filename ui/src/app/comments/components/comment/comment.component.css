.comments-container {
	margin: 0 24px;
}

.comment-card {
	width: 100%;
  margin-top: 1em;
}

.comment-resolved {
	opacity: 0.7;
}

.reply-card {
	margin: 0 24px;
}

med-comment > .comments-container {
	margin: 0;
}

med-comment > .reply-card,
.comments-container > .reply-card {
	margin: 0;
}

mat-card-content {
  overflow-wrap: break-word;
}



med-comment > .comments-container > .comment-card {
	background-color: #F0F0F0;
	width: 100%;
  margin-top: 0;
}

.reply-form-field {
	width: 100%;
}

.comment-header-image {
	background-image: url('https://i.imgflip.com/cnudu.jpg');
  background-size: cover;
}

.comment-created-by {
  margin-bottom: 5px;
}

.comment-created-on {
  font-size: 10px;
}

.comment-text {
  margin-bottom: 0;
}

