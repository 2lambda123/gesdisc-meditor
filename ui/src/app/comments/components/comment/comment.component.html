<div *ngFor="let comment of extComments" class="comments-container">
	<mat-card class="comment-card" 
    *ngIf="_showResolved || !comment.resolved" 
    [class.comment-resolved]="comment.resolved">
		<mat-card-header>
			<div mat-card-avatar class="comment-header-image"></div>
			<mat-card-title class="comment-created-by">{{comment.createdBy}}</mat-card-title>
			<mat-card-subtitle class="comment-created-on">{{comment.createdOn | date: 'short' }}</mat-card-subtitle>
		</mat-card-header>

		<mat-card-content class="comment-text">
			{{comment.text}}
		</mat-card-content>

		<mat-card-actions *ngIf="!comment.resolved">
			<button mat-icon-button class="reply-button"
				(click)="openReplyForm(comment._id)">
				<mat-icon>reply</mat-icon>
			</button>
			<button mat-icon-button class="resolve-button"
        *ngIf="comment.parentId == 'root'"
				(click)="resolve(comment._id)">
				<mat-icon>done</mat-icon>
			</button>
		</mat-card-actions>
	</mat-card>
	<mat-card class="reply-card" *ngIf="comment.replyTo">
    <mat-card-content>
      <mat-form-field class="reply-form-field">
        <textarea matInput #commentForm placeholder="Comment" [(ngModel)]="commentText"></textarea>
      </mat-form-field>
     </mat-card-content>
	  <mat-card-actions>
	    <button mat-icon-button class="action-button" (click)="submitComment(comment._id, comment.parentId)">
	      <mat-icon>send</mat-icon>
	    </button>
	    <button mat-icon-button class="action-button" (click)="closeReply(comment._id)">
	      <mat-icon>cancel</mat-icon>
	    </button>
	  </mat-card-actions>
  </mat-card>
	<med-comment *ngIf="comment.children" #commentThread
  	[comments]="comment.children"
  	[tree]="false"
  	[showResolved]="_showResolved"
    [parentId] = "comment._id"
  	(resolveComment)="resolve($event)"
		(replyComment)="submitChildComment($event)">
  </med-comment>
</div>
