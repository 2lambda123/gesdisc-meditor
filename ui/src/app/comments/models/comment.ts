export class Comment {
	constructor(
	  public _id: string,
	  public _parentId: string,
	  public author: string,
	  public text: string,
	  public deleted: boolean,
	  public visible: boolean,
	  public replyTo: boolean,
	  public children: [Comment]
	) {};
}
