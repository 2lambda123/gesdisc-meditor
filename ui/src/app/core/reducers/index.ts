import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
