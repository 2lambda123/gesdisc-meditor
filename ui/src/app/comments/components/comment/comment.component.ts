import { Component, OnInit, Input } from '@angular/core';
// import { CommonService } from '../../common.service';
import { Observable } from 'rxjs/Rx';
import { FormControl, FormGroup, FormBuilder, Validators, NgForm } from '@angular/forms';
import { Comment } from '../../models/comment';

import * as _ from 'underscore';

@Component({
  selector: 'med-comment',
  templateUrl: './comment.component.html',
  styleUrls: ['./comment.component.css']
})
export class CommentComponent implements OnInit {

	@Input() comments: Comment[];

  commentForm: FormGroup;
  author:string='';
  comment:string='';

  constructor(private fb: FormBuilder) {
  	this.commentForm = fb.group({
	  	'author': [null, Validators.required],
	  	'text': [null, Validators.required],
	  	'_parentId': [null, Validators.required]
	  })
  }

  ngOnInit() {
  }

  onFormSubmit(form: any) {
  	let comment;
  	comment = {
  		'author': form.author,
  		'text': form.text,
  		'_parentId': form._parentId
  	}
  	// this.commentService.create(comment).subscribe(
   //    data => {
   //    	let parentComment  = _.find(this.comments, function(comment) { return comment._id == form._parentId; });
   //    	parentComment.replyTo = false;
			// 	comment.children = [];
			// 	parentComment.children.push(comment);
   //    },
   //    err => console.error(err),
   //    () => { console.log('done creating new comment') }
   //  );

  }

  replyTo(id, parentId) {
  	_.find(this.comments, function(comment) { return comment._id == id }).replyTo = true;
  }

  delete(id) {
  	// this.commentService.delete(id).subscribe(
  	// 	data => { _.find(this.comments, function(comment) { return comment._id == id }).deleted = true; },
   //    err => console.error(err),
   //    () => { }
   //  );
  }

  showReplies(id, parentId) {
  	let comment = _.find(this.comments, function(comment) { return comment._id == id });
  	if(comment.visible) {
  		_.find(this.comments, function(comment) { return comment._id == id }).visible = false;
  	} else {
  		_.find(this.comments, function(comment) { return comment._id == id }).visible = true;
  	}
  }

}
