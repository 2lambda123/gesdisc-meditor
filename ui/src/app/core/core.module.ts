import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
