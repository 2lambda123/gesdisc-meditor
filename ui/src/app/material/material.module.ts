import { NgModule } from '@angular/core';

import {
  MatInputModule,
  MatCardModule,
  MatButtonModule,
  MatSidenavModule,
  MatListModule,
  MatIconModule,
  MatToolbarModule,
  MatProgressSpinnerModule,
  MatDatepickerModule,
  MatCheckboxModule,
  MatSelectModule
} from '@angular/material';

@NgModule({
  imports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
  exports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
})
export class MaterialModule {}
