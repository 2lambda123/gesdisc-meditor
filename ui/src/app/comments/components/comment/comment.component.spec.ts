import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { MatCardModule, MatIconModule, MatInputModule } from '@angular/material';

import { CommentComponent } from './comment.component';

describe('CommentComponent', () => {
  let component: CommentComponent;
  let fixture: ComponentFixture<CommentComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [ 
        NoopAnimationsModule,
        FormsModule,
        MatCardModule,
        MatIconModule,
        MatInputModule
      ],
      declarations: [ CommentComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CommentComponent);
    component = fixture.componentInstance;
  });

  it('should create comment thread with mock data', () => {
    component.tree = true;
    component.comments = 
      [
        {
          _id: '5bc9fed4e7cd1d776e763685',
          text: 'test',
          resolved: false,
          parentId: 'root',
          documentId: 'test',
          model: 'Image',
          createdBy: 'Test',
          createdOn: '2018-10-19T15:57:08.084Z'
        },
        {
          _id: '5bca04bace5e5478ac9c985f',
          text: 'image comment',
          resolved: false,
          parentId: 'root',
          documentId: 'test',
          model: 'Image',
          createdBy: 'Test',
          createdOn: '2018-10-19T16:22:18.565Z'
        },
        {
          _id: '5bcf41e2e59cd2e18508b1e7',
          text: 'wow another reply here',
          resolved: false,
          parentId: '5bc9fed4e7cd1d776e763685',
          documentId: 'test',
          model: 'Image',
          createdBy: 'Test',
          createdOn: '2018-10-23T15:44:34.601Z'
        },
        {
          _id: '5bcf4208e59cd2e18508b1e8',
          text: 'another important reply',
          resolved: false,
          parentId: '5bc9fed4e7cd1d776e763685',
          documentId: 'test',
          model: 'Image',
          createdBy: 'Test',
          createdOn: '2018-10-23T15:45:12.535Z'
        },
        {
          _id: '5bcf421be59cd2e18508b1e9',
          text: 'hey topic starter, pay attention to this please',
          resolved: false,
          parentId: '5bc9fed4e7cd1d776e763685',
          documentId: 'test',
          model: 'Image',
          createdBy: 'Test',
          createdOn: '2018-10-23T15:45:31.108Z'
        }
      ]
    
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
