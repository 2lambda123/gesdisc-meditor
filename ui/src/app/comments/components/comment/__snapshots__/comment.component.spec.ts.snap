// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentComponent should create comment thread with mock data 1`] = `
<med-comment
  _tree={[Function Boolean]}
  commentForm="undefined"
  commentThreads={[Function QueryList]}
  extComments={[Function Array]}
  replyComment={[Function EventEmitter]}
  resolveComment={[Function EventEmitter]}
>
  <div
    class="comments-container"
  >
    
    <mat-card
      class="comment-card mat-card"
    >
      <mat-card-header
        class="mat-card-header"
      >
        <div
          class="comment-header-image mat-card-avatar"
          mat-card-avatar=""
        />
        <div
          class="mat-card-header-text"
        >
          <mat-card-title
            class="comment-created-by mat-card-title"
          >
            Test
          </mat-card-title>
          <mat-card-subtitle
            class="comment-created-on mat-card-subtitle"
          >
            10/19/18, 11:57 AM
          </mat-card-subtitle>
        </div>
      </mat-card-header>
      <mat-card-content
        class="comment-text mat-card-content"
      >
         test 
      </mat-card-content>
      
      <mat-card-actions
        class="mat-card-actions"
      >
        <button
          class="reply-button"
          mat-icon-button=""
        >
          <mat-icon
            aria-hidden="true"
            class="mat-icon material-icons"
            role="img"
          >
            reply
          </mat-icon>
        </button>
        
        <button
          class="resolve-button"
          mat-icon-button=""
        >
          <mat-icon
            aria-hidden="true"
            class="mat-icon material-icons"
            role="img"
          >
            done
          </mat-icon>
        </button>
      </mat-card-actions>
    </mat-card>
    
    
    <med-comment
      ng-reflect-comments="[object Object],[object Object"
      ng-reflect-parent-id="5bc9fed4e7cd1d776e763685"
      ng-reflect-tree="false"
    >
      
      <div
        class="comments-container"
      >
        
        <mat-card
          class="comment-card mat-card"
        >
          <mat-card-header
            class="mat-card-header"
          >
            <div
              class="comment-header-image mat-card-avatar"
              mat-card-avatar=""
            />
            <div
              class="mat-card-header-text"
            >
              <mat-card-title
                class="comment-created-by mat-card-title"
              >
                Test
              </mat-card-title>
              <mat-card-subtitle
                class="comment-created-on mat-card-subtitle"
              >
                10/23/18, 11:44 AM
              </mat-card-subtitle>
            </div>
          </mat-card-header>
          <mat-card-content
            class="comment-text mat-card-content"
          >
             wow another reply here 
          </mat-card-content>
          
          <mat-card-actions
            class="mat-card-actions"
          >
            <button
              class="reply-button"
              mat-icon-button=""
            >
              <mat-icon
                aria-hidden="true"
                class="mat-icon material-icons"
                role="img"
              >
                reply
              </mat-icon>
            </button>
            
          </mat-card-actions>
        </mat-card>
        
        
        <med-comment
          ng-reflect-comments=""
          ng-reflect-parent-id="5bcf41e2e59cd2e18508b1e7"
          ng-reflect-tree="false"
        >
          
        </med-comment>
      </div>
      <div
        class="comments-container"
      >
        
        <mat-card
          class="comment-card mat-card"
        >
          <mat-card-header
            class="mat-card-header"
          >
            <div
              class="comment-header-image mat-card-avatar"
              mat-card-avatar=""
            />
            <div
              class="mat-card-header-text"
            >
              <mat-card-title
                class="comment-created-by mat-card-title"
              >
                Test
              </mat-card-title>
              <mat-card-subtitle
                class="comment-created-on mat-card-subtitle"
              >
                10/23/18, 11:45 AM
              </mat-card-subtitle>
            </div>
          </mat-card-header>
          <mat-card-content
            class="comment-text mat-card-content"
          >
             another important reply 
          </mat-card-content>
          
          <mat-card-actions
            class="mat-card-actions"
          >
            <button
              class="reply-button"
              mat-icon-button=""
            >
              <mat-icon
                aria-hidden="true"
                class="mat-icon material-icons"
                role="img"
              >
                reply
              </mat-icon>
            </button>
            
          </mat-card-actions>
        </mat-card>
        
        
        <med-comment
          ng-reflect-comments=""
          ng-reflect-parent-id="5bcf4208e59cd2e18508b1e8"
          ng-reflect-tree="false"
        >
          
        </med-comment>
      </div>
      <div
        class="comments-container"
      >
        
        <mat-card
          class="comment-card mat-card"
        >
          <mat-card-header
            class="mat-card-header"
          >
            <div
              class="comment-header-image mat-card-avatar"
              mat-card-avatar=""
            />
            <div
              class="mat-card-header-text"
            >
              <mat-card-title
                class="comment-created-by mat-card-title"
              >
                Test
              </mat-card-title>
              <mat-card-subtitle
                class="comment-created-on mat-card-subtitle"
              >
                10/23/18, 11:45 AM
              </mat-card-subtitle>
            </div>
          </mat-card-header>
          <mat-card-content
            class="comment-text mat-card-content"
          >
             hey topic starter, pay attention to this please 
          </mat-card-content>
          
          <mat-card-actions
            class="mat-card-actions"
          >
            <button
              class="reply-button"
              mat-icon-button=""
            >
              <mat-icon
                aria-hidden="true"
                class="mat-icon material-icons"
                role="img"
              >
                reply
              </mat-icon>
            </button>
            
          </mat-card-actions>
        </mat-card>
        
        
        <med-comment
          ng-reflect-comments=""
          ng-reflect-parent-id="5bcf421be59cd2e18508b1e9"
          ng-reflect-tree="false"
        >
          
        </med-comment>
      </div>
    </med-comment>
  </div><div
    class="comments-container"
  >
    
    <mat-card
      class="comment-card mat-card"
    >
      <mat-card-header
        class="mat-card-header"
      >
        <div
          class="comment-header-image mat-card-avatar"
          mat-card-avatar=""
        />
        <div
          class="mat-card-header-text"
        >
          <mat-card-title
            class="comment-created-by mat-card-title"
          >
            Test
          </mat-card-title>
          <mat-card-subtitle
            class="comment-created-on mat-card-subtitle"
          >
            10/19/18, 12:22 PM
          </mat-card-subtitle>
        </div>
      </mat-card-header>
      <mat-card-content
        class="comment-text mat-card-content"
      >
         image comment 
      </mat-card-content>
      
      <mat-card-actions
        class="mat-card-actions"
      >
        <button
          class="reply-button"
          mat-icon-button=""
        >
          <mat-icon
            aria-hidden="true"
            class="mat-icon material-icons"
            role="img"
          >
            reply
          </mat-icon>
        </button>
        
        <button
          class="resolve-button"
          mat-icon-button=""
        >
          <mat-icon
            aria-hidden="true"
            class="mat-icon material-icons"
            role="img"
          >
            done
          </mat-icon>
        </button>
      </mat-card-actions>
    </mat-card>
    
    
    <med-comment
      ng-reflect-comments=""
      ng-reflect-parent-id="5bca04bace5e5478ac9c985f"
      ng-reflect-tree="false"
    >
      
    </med-comment>
  </div>
</med-comment>
`;
